sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/SessionCatalog.scala

```diff
@@ -368,6 +368,12 @@ class SessionCatalog(
       tableName: TableIdentifier,
       parts: Seq[CatalogTablePartition],
       ignoreIfExists: Boolean): Unit = {
+    val tableMetadata = getTableMetadata(tableName)
+    // Verify if each part exactly matches the partition spec of the table definition
+    parts.foreach { p =>
```

Comment:
We do not allow it. Here, `requireExactMatchedPartitionSpec ` is to verify if two sequences are exactly identical.